### ISSUE TITLE
stdlib_*.nim.c files generated incompatible with my platform

### ISSUE THREAD

**Issue by zevv** [nim-works/nimskull#1190]
Trying to compile a simple program for the `avr` architecture; works fine with nim, but with nimskull I end up with a handful of `stdlib_*.nim.c` files, one of which includes a header that 's not available for this particular architecture:

```
/home/ico/.cache/nimskull/main_r/stdlib_fenv.nim.c:8:10: fatal error: fenv.h: No such file or directory
 #include <fenv.h>
          ^~~~~~~~
compilation terminated.
```
I have no short and simple repro, but the code showing the problem is at https://github.com/zevv/cps-baremetal/tree/master

**Comment by saem** [comment]
I'm not against supporting the use case, as long as we can actually verify it on an ongoing basis and the hit to the backend is minimal/non-existent.

~I'm not really sure how we might approach that, but if you have some ideas we can start developing an approach to at least discuss the merits of.~

Update:

Testing can be accomplished if we can target `avr` and then use `simavr` in CI to at least have a basic canary.

**Comment by zerbina** [comment]
Fixed by #1191, but wasn't auto-closed. If not using any of the globals from `fenv`, the `fenv.h` header is no longer included.
